File: DemoApi/src/InMemoryDb.Model/Product.cs
using Newtonsoft.Json;

using Application.Framework;

namespace InMemoryDb.Model;

public record Product : BaseModel<Guid>
{
    public int ProductId { get; init; }
    public string ProductName { get; init; }
    public string ProductCode { get; init; }
    internal string _tagList;
    public string[]? TagList
    {
        get { return string.IsNullOrEmpty(_tagList) ? null : JsonConvert.DeserializeObject<string[]>(_tagList); }
        set { _tagList = JsonConvert.SerializeObject(value); }
    }
    public DateTime ReleaseDate { get; init; }
    public double Price { get; init; }
    public string Description { get; init; }
    public double StarRating { get; init; }
    public string ImageUrl { get; init; }
}

File: Angular/src/DemoApp/src/app/demo/model/product.ts
import { BaseEntity } from '@common/model/baseEntity';

export class Product extends BaseEntity {
  productId: number;
  productName: string;
  productCode: string;
  tagList?: string[];
  releaseDate: Date;
  price: number;
  description: string;
  starRating: number;
  imageUrl: string;
}
